File: apps/api/app/domain/enums.py
"""Enum di dominio condivisi.

I valori (stringhe) sono il contratto effettivo sul filo: devono combaciare 1:1 con i
literal union types definiti in `packages/shared`.
"""

from enum import Enum


class TryOnMode(str, Enum):
    """Modalita' d'uso del prodotto."""

    PHOTO_MEASUREMENTS = "photo_measurements"  # foto utente + misure
    MEASUREMENTS_ONLY = "measurements_only"  # solo misure -> avatar rappresentativo


class JobStatus(str, Enum):
    """Stato del ciclo di vita di un job di try-on."""

    QUEUED = "queued"
    PROCESSING = "processing"
    DONE = "done"
    FAILED = "failed"
    CANCELED = "canceled"


class GarmentCategory(str, Enum):
    """Categoria del capo, rilevante per il try-on."""

    TOP = "top"
    BOTTOM = "bottom"
    DRESS = "dress"
    OUTERWEAR = "outerwear"
    OTHER = "other"


class MeasurementUnit(str, Enum):
    """Sistema di unita' delle misure corporee."""

    METRIC = "metric"  # cm / kg
    IMPERIAL = "imperial"  # in / lb


class FitPreference(str, Enum):
    """Preferenza di vestibilita' dichiarata dall'utente (separata dai dati corporei)."""

    SLIM = "slim"
    REGULAR = "regular"
    RELAXED = "relaxed"


class TryOnQuality(str, Enum):
    """Livello qualitativo del rendering richiesto al provider."""

    PREVIEW = "preview"
    STANDARD = "standard"
    HIGH = "high"

File: packages/shared/src/enums.ts
/**
 * Enum di dominio condivisi (literal union types).
 *
 * I valori stringa sono il contratto effettivo sul filo e devono combaciare 1:1
 * con gli enum Pydantic in `apps/api/app/domain/enums.py`.
 */

export const TryOnMode = {
  PHOTO_MEASUREMENTS: "photo_measurements",
  MEASUREMENTS_ONLY: "measurements_only",
} as const;
export type TryOnMode = (typeof TryOnMode)[keyof typeof TryOnMode];

export const JobStatus = {
  QUEUED: "queued",
  PROCESSING: "processing",
  DONE: "done",
  FAILED: "failed",
  CANCELED: "canceled",
} as const;
export type JobStatus = (typeof JobStatus)[keyof typeof JobStatus];

export const GarmentCategory = {
  TOP: "top",
  BOTTOM: "bottom",
  DRESS: "dress",
  OUTERWEAR: "outerwear",
  OTHER: "other",
} as const;
export type GarmentCategory = (typeof GarmentCategory)[keyof typeof GarmentCategory];

export const MeasurementUnit = {
  METRIC: "metric",
  IMPERIAL: "imperial",
} as const;
export type MeasurementUnit = (typeof MeasurementUnit)[keyof typeof MeasurementUnit];

export const FitPreference = {
  SLIM: "slim",
  REGULAR: "regular",
  RELAXED: "relaxed",
} as const;
export type FitPreference = (typeof FitPreference)[keyof typeof FitPreference];

export const TryOnQuality = {
  PREVIEW: "preview",
  STANDARD: "standard",
  HIGH: "high",
} as const;
export type TryOnQuality = (typeof TryOnQuality)[keyof typeof TryOnQuality];
